### PR TITLE
fix some deploy issues and update registration pkg version to latest

### PR DESCRIPTION
**Pull request by zhiweiyin318**
Signed-off-by: Zhiwei Yin <zyin@redhat.com>
1. fix the work-manager addon miss agent clusterrole.
2. fix the readme typo.
3. fix the demo script issue. 
4. update the registration pkg to the latest which includes some fixes.

### CODE REVIEW

File: controlplane/hack/demo/demo-magic.sh [hunk 1]
```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env bash
```
Comment (clyang82) [inline controlplane/hack/demo/demo-magic.sh:1]: for third-party lib, it is better to reference it

Comment (zhiweiyin318) [reply controlplane/hack/demo/demo-magic.sh:1]: ok, removed.

**Pull request merged**